### ISSUE TITLE
Incorrect SSL ca, key or cert data causes uncaughtException

### ISSUE THREAD

**Issue by ianfairbairn** [brianc/node-postgres#2004]
Description:

I can successfully connect to a postgreSQL database using SSL if the certs (as strings) are correctly entered.  However, if I alter the ca, key or cert strings (to replicate a user fat-fingering them as they cut and paste them in the UI) I get an uncaughtException and my service falls over because the error doesn't bubble up to me to handle it.

Env:
Docker container
nodejs v11.6.0
pg v^7.12.0

Code:

const postgreSQL = require('pg');

exports.execSQL = (databaseConnectionModel, sql) => new Promise((resolve, reject) => {
	
	try {

		let pool;
		let config = {};

		config = {
			host: databaseConnectionModel.host,
			port: databaseConnectionModel.port,
			user: databaseConnectionModel.user,
			password: databaseConnectionModel.password,
			database: databaseConnectionModel.database
		}

		if (databaseConnectionModel.useSSL) {
			config.ssl = {
				ca: databaseConnectionModel.ca,
				key: databaseConnectionModel.key,
				cert: databaseConnectionModel.cert
			}
		}

		pool = new postgreSQL.Pool(config);

		pool.query(sql, (error, result) => {

			if (error) {           
				pool.end();
				reject(error); 
			} else {    
				pool.end();
				resolve(result);
			}
		});

	} catch (error) {
		reject(error);
	}
});

error:

error: uncaughtException: error:0906D064:PEM routines:PEM_read_bio:bad base64 decode
Error: error:0906D064:PEM routines:PEM_read_bio:bad base64 decode
    at Object.createSecureContext (_tls_common.js:134:17)
    at Object.connect (_tls_wrap.js:1221:48)
    at Socket.<anonymous> (/usr/src/app/node_modules/pg/lib/connection.js:95:23)
    at Object.onceWrapper (events.js:276:13)
    at Socket.emit (events.js:188:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17) {"error":{},"stack":"Error: error:0906D064:PEM routines:PEM_read_bio:bad base64 decode\n    at Object.createSecureContext (_tls_common.js:134:17)\n    at Object.connect (_tls_wrap.js:1221:48)\n    at Socket.<anonymous> (/usr/src/app/node_modules/pg/lib/connection.js:95:23)\n    at Object.onceWrapper (events.js:276:13)\n    at Socket.emit (events.js:188:13)\n    at addChunk (_stream_readable.js:288:12)\n    at readableAddChunk (_stream_readable.js:269:11)\n    at Socket.Readable.push (_stream_readable.js:224:10)\n    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)","exception":true,"date":"Fri Nov 08 2019 05:02:33 GMT+0000 (Coordinated Universal Time)","process":{"pid":17,"uid":0,"gid":0,"cwd":"/usr/src/app","execPath":"/usr/local/bin/node","version":"v11.6.0","argv":["/usr/local/bin/node","/usr/src/app/index.js"],"memoryUsage":{"rss":51814400,"heapTotal":20148224,"heapUsed":15393352,"external":58506}},"os":{"loadavg":[0.078125,0.05615234375,0.0927734375],"uptime":25640},"trace":[{"column":17,"file":"_tls_common.js","function":"Object.createSecureContext","line":134,"method":"createSecureContext","native":false},{"column":48,"file":"_tls_wrap.js","function":"Object.connect","line":1221,"method":"connect","native":false},{"column":23,"file":"/usr/src/app/node_modules/pg/lib/connection.js","function":null,"line":95,"method":null,"native":false},{"column":13,"file":"events.js","function":"Object.onceWrapper","line":276,"method":"onceWrapper","native":false},{"column":13,"file":"events.js","function":"Socket.emit","line":188,"method":"emit","native":false},{"column":12,"file":"_stream_readable.js","function":"addChunk","line":288,"method":null,"native":false},{"column":11,"file":"_stream_readable.js","function":"readableAddChunk","line":269,"method":null,"native":false},{"column":10,"file":"_stream_readable.js","function":"Socket.Readable.push","line":224,"method":"push","native":false},{"column":17,"file":"internal/stream_base_commons.js","function":"TCP.onStreamRead [as onread]","line":145,"method":"onStreamRead [as onread]","native":false}],"timestamp":"08-11-2019 05:02:33"}

**Comment by YegorZaremba** [comment]
according [test](https://github.com/brianc/node-postgres/blob/master/packages/pg/test/unit/connection-parameters/creation-tests.js#L319) you should use following name of keys:
```
ssl: {
      sslca: databaseConnectionModel.ca,
      sslkey: databaseConnectionModel.key,
      sslcert: databaseConnectionModel.cert
},
```

**Comment by boromisp** [comment]
The cause of the issue is that [`tls.connect(...)`](https://github.com/brianc/node-postgres/blob/ea6ac2ad2313af57159b10a0292c0c178e8e0923/packages/pg/lib/connection.js#L92) can throw.

How to reproduce:

```js
const { Pool } = require('pg');

const pool = new Pool({
  ssl: {
    ca: 'x', key: 'y', cert: 'z',
    rejectUnauthorized: false
  }
});

(async () => {
  try {
    await pool.query('SELECT 1');
  } catch (error) {
    console.error('CATCH', error);
  }
})();
```

Wrapping the `.connect()` in try - catch seems to work, but I'm not sure if there is more to clean up (what happens with the original `this.stream`?):

```js
try {
  self.stream = tls.connect(options)
} catch (err) {
  reportStreamError(err)
  this.emit('end')
  return
}
```

**Comment by chris--young** [comment]
Can I get someone to checkout the pr I opened to fix this? This bug is causing an issue for my project and I would rather fix the problem at the source than to implement a workaround.

**Comment by charmander** [comment]
I wish it could throw synchronously when creating the client or (especially) pool, but it doesn’t look like there’s a convenient way to achieve that with `tls.connect` =/

**Comment by chris--young** [comment]
Yea, that isn't how it works. There's no way to do that without major breaking changes. For sure it is less than ideal that `tls.connect()` throws, but I can't imagine fixing it this way is worse than leaving a known fatal bug in the project.

**Comment by brianc** [comment]
published fix (thank you @chris--young!) at `pg@8.3.2`
